protobuf/protoc-gen-gogrpcmock/plugin/plugin.go

```diff
@@ -230,48 +248,150 @@ func isSupportedInt(field *descriptor.FieldDescriptorProto) bool {
 	return false
 }
 
-func generateStringValue(fieldName string) string {
-	val := ""
+func generateStringValue(fieldName string, field *descriptor.FieldDescriptorProto) string {
+
+	if mocks := getFieldMocksIfAny(field); mocks != nil {
+		var sb strings.Builder
+
+		if len(mocks.Prefix) > 0 {
+			sb.WriteString(mocks.Prefix[r.Intn(len(mocks.Prefix))])
+		}
+
+		if len(mocks.String_) > 0 {
+			sb.WriteString(mocks.String_[r.Intn(len(mocks.String_))])
+		}
+
+		if boolFromPtr(mocks.Word) {
+			sb.WriteString(fake.Word())
+		}
+
+		if boolFromPtr(mocks.Words) {
+			sb.WriteString(fake.Words())
+		}
+
+		if mocks.Wordsn != nil {
+			sb.WriteString(fake.WordsN(int(*mocks.Wordsn)))
+		}
+
+		if boolFromPtr(mocks.Fullname) {
+			sb.WriteString(fake.FullName())
+		}
+
+		if boolFromPtr(mocks.Firstname) {
```

Comment:
Since we aren't returning in this function, seems like multiple of these conditions can be applied at the same time. For e.g. firstname, lastname. I am assuming this is by design?

Reply:
Yes, By design. Users can string the output they desire.